File: Mission 8/src/test/java/com/izzaz/mission8/service/ServiceTest.java
package com.izzaz.mission8.service;

import com.izzaz.mission8.entity.Game;
import com.izzaz.mission8.repository.GameRepository;
import org.junit.*;
import org.junit.runner.RunWith;
import org.mockito.*;
import org.mockito.junit.MockitoJUnitRunner;
import java.util.*;

import static org.junit.jupiter.api.Assertions.*;

@RunWith(MockitoJUnitRunner.class)
public class ServiceTest {

    @InjectMocks
    private GameService gameService;

    @Mock
    private GameRepository gameRepository;

    @Test
    public void getGameListTest(){
        Mockito.when(gameRepository.getGameList()).thenReturn(Arrays.asList(
           new Game(1,"GTA5","PC"),
           new Game(2,"NFS:Carbon","PS2")
        ));
        List<Game> gameList = gameService.getGameList();
        assertEquals(1, gameList.get(0).getId());
        assertEquals("GTA5", gameList.get(0).getName());
        assertEquals("PC", gameList.get(0).getPlatform());
        assertEquals(2, gameList.get(1).getId());
        assertEquals("NFS:Carbon", gameList.get(1).getName());
        assertEquals("PS2", gameList.get(1).getPlatform());

    }



    @Test
    public void getGameTest(){
        Mockito.when(gameRepository.getGame(1)).thenReturn(new Game(1,"GTA5","PC"));
        Game game = gameService.getGame(1);
        assertEquals(1, game.getId());
        assertEquals("GTA5", game.getName());
        assertEquals("PC", game.getPlatform());
    }

    @Test
    public void getEmptyGameTest(){
        Mockito.when(gameRepository.getGame(1)).thenReturn(null);
        Game game = gameService.getGame(1);
        assertNull(game);
    }

}

File: kotlinsb/src/test/kotlin/com/izzaz/kotlinsb/service/ServiceTest.kt
package com.izzaz.kotlinsb.service

import com.izzaz.kotlinsb.entity.Game
import com.izzaz.kotlinsb.repository.GameRepo
import junit.framework.Assert.assertEquals
import junit.framework.Assert.assertNull
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.Mockito.`when`
import org.mockito.junit.MockitoJUnitRunner
import org.springframework.boot.test.context.SpringBootTest

@RunWith(MockitoJUnitRunner::class)
@SpringBootTest
class ServiceTest() {

    @InjectMocks
    private lateinit var gameService: GameService

    @Mock
    private lateinit var gameRepo: GameRepo

    @Test
    fun getGameListTest() {
        `when`(gameRepo.getGameList()).thenReturn(listOf(
                Game(10,"GTA5","PC"),
                Game(11,"Forza : Horizon","Xbox")
        ))
        val gameList : List<Game> = gameService.getGameList()
        assertEquals(10, gameList[0].id)
        assertEquals("GTA5", gameList[0].name)
        assertEquals("PC", gameList[0].platform)
        assertEquals(11, gameList[1].id)
        assertEquals("Forza : Horizon", gameList[1].name)
        assertEquals("Xbox", gameList[1].platform)
    }

    @Test
    fun getGameTest(){
        `when`(gameRepo.getGame(10)).thenReturn(Game(10,"GTA5","PC"))
        val game : Game? = gameService.getGame(10)
        assertEquals(10, game?.id)
        assertEquals("GTA5", game?.name)
        assertEquals("PC", game?.platform)
    }

    @Test
    fun getEmptyGameTest(){
        `when`(gameRepo.getGame(1)).thenReturn(null)
        val game : Game? = gameService.getGame(1)
        assertNull(game)
    }

}